massgov/mayflower
# React/DP-14476: Adds support for writing tests using jest/enzyme/storybook-jest.

## Description
  This adds support for testing with jest/enzyme within storybook. Jest tests are ran and watched with `react-scripts test`, which allows for tests to also be ran through babel for transpiling. Test results are written to a file under the root directory `.jest-test-results.json`. The configuration file for tests is located under the root directory as `setupTests.js`.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-14476)

## Steps to Test
1. Pull the branch and run: `nvm use` `npm install` `npm run start`.
2. Verify that jest is running in watch mode by changing anything in the project files under src. You should see the watch terminal screen re-run tests.
3. Browse to atoms -> ArrowButton and Button and verify that you see a `Jest` tab  similar to the screenshot below, and the test there is labeled as passing.

## Screenshots
<img width="1465" alt="Screen Shot 2019-06-25 at 3 43 45 PM" src="https://user-images.githubusercontent.com/5257660/60130281-f259b580-9764-11e9-9233-09753ed252d1.png">

## Additional Notes:
When adding tests to jest within your story file, use the whole name of the test file.
